tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/ensure_date_datetime_consistency_test.py

```diff
@@ -0,0 +1,178 @@
+the date and datetime field, the date field value is preferred.  If a time
+is provided in the field, the time is maintained.  If the datetime field
+was null, a default value of midnight is used for the time component. 
+"""
+# Python imports
+from datetime import date
+import os
+
+# Third party imports
+from dateutil import parser
+from jinja2 import Template
+
+# Project imports
+from app_identity import PROJECT_ID
+from cdr_cleaner.cleaning_rules.ensure_date_datetime_consistency import EnsureDateDatetimeConsistency, TABLE_DATES
+from tests.integration_tests.data_steward.cdr_cleaner.cleaning_rules.bigquery_tests_base import BaseTest
+
+
+class EnsureDateDatetimeConsistencyTest(BaseTest.CleaningRulesTestBase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+        super().initialize_class_vars()
+
+        # set the test project identifier
+        project_id = os.environ.get(PROJECT_ID)
+        cls.project_id = project_id
+
+        # set the expected test datasets
+        dataset_id = os.environ.get('COMBINED_DATASET_ID')
+        sandbox_id = dataset_id + '_sandbox'
+
+        cls.query_class = EnsureDateDatetimeConsistency(project_id, dataset_id,
+                                                        sandbox_id)
+
+        sb_table_names = cls.query_class.get_sandbox_tablenames()
+        for table_name in sb_table_names:
+            cls.fq_sandbox_table_names.append(
+                f'{project_id}.{sandbox_id}.{table_name}')
+
+        for key in TABLE_DATES:
+            cls.fq_table_names.append(f"{project_id}.{dataset_id}.{key}")
+
+        # call super to set up the client, create datasets, and create
+        # empty test tables
+        # NOTE:  does not create empty sandbox tables.
+        super().setUpClass()
+
+    def setUp(self):
+        """
+        Create common information for tests.
+
+        Creates common expected parameter types from cleaned tables and
+        a common fully qualified dataset name string used to load
+        the data.
+        """
+        self.start_date = parser.parse('2016-05-01').date()
+        self.start_datetime = parser.parse('2016-05-01 11:00:00 UTC')
+        self.start_default_datetime = parser.parse('2016-05-01 00:00:00 UTC')
+        self.end_date = parser.parse('2016-05-02').date()
+        self.end_datetime = parser.parse('2016-05-02 11:00:00 UTC')
+        self.end_default_datetime = parser.parse('2016-05-02 00:00:00 UTC')
+
+        fq_dataset_name = self.fq_table_names[0].split('.')
+        self.fq_dataset_name = '.'.join(fq_dataset_name[:-1])
+
+    def test_nullable_field_datetime_pair(self):
+        """
+        Use the default test function.
+
+        Validates pre-conditions, test execution and post conditions based on
```

Comment:
opt_nit: This comment sentence isn't doing much for me, in comparison to just reading the code - feels like something that might get copy-pasted around a lot as boilerplate. I would probably remove.

Reply:
updated both method doc strings.